src/external_task_api_service.ts

```diff
@@ -129,6 +128,43 @@ export class ExternalTaskApiService implements IExternalTaskApi {
     this._publishExternalTaskFinishedMessage(externalTask, successNotificationPayload);
   }
 
+  private async _fetchOrWaitForExternalTasks<TPayloadType>(topicName: string, maxTasks: number, longPollingTimeout: number)
+    : Promise<Array<ExternalTask<TPayloadType>>> {
+
+    return new Promise<Array<ExternalTask<TPayloadType>>>(async (resolve): Promise<void> => {
+
+      const tasks: Array<ExternalTask<TPayloadType>> =
+        await this._externalTaskRepository.fetchAvailableForProcessing<TPayloadType>(topicName, maxTasks);
+
+      if (tasks.length > 0) {
+        resolve(tasks);
+      }
+      let subscription: ISubscription;
+
+      const timeout = setTimeout(() => {
+        subscription.dispose();
+        resolve([]);
+      },
+        longPollingTimeout
+      );
+
+      const externalTaskCreatedEventName = `/externaltask/topic/${topicName}/created`;
+      subscription = this._eventAggregator.subscribeOnce(externalTaskCreatedEventName, async (): Promise<void> => {
+
+        clearTimeout(timeout);
+
+        const tasks: Array<ExternalTask<TPayloadType>> =
+          await this._externalTaskRepository.fetchAvailableForProcessing<TPayloadType>(topicName, maxTasks);
+
+        if (tasks.length > 0) {
```

Comment:
Extract business rule like stated above.

```suggestion
        const tasksAreNotEmpty: boolean = tasks.length > 0;
        if (tasksAreNotEmpty) {
```